jerry-ext/private-headers/section-utils.h

```diff
@@ -0,0 +1,67 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SECTION_UTILS_H
+#define SECTION_UTILS_H
+
+/**
+ * Declare references to a section that contains an array of items.
+ *
+ * @name the name of the section (without quotes)
+ * @type the type of the elements stored in the array
+ *
+ * This macro needs to be placed before any occurrence of ITERATE_OVER_SECTION, for a given section.
+ */
+#ifdef __MACH__
+#define DECLARE_SECTION(name, type)                                  \
+extern type __start_ ## name[] __asm("section$start$__DATA$" #name); \
+extern type __stop_ ## name[] __asm("section$end$__DATA$" #name);
+#else /* !__MACH__ */
+#define DECLARE_SECTION(name, type)                   \
+extern type __start_ ## name[] __attribute__((weak)); \
+extern type __stop_ ## name[] __attribute__((weak));
+#endif /* __MACH__ */
+
+/**
+ * Iterate over items in an array stored in a section
+ *
+ * @name is the name of the section.
```

Comment:
AFAIK this is not proper doxygen syntax. You can describe function/macro parameters by
```
@param name description till the end of the line
```
and you can reference them anywhere in the text by `bla bla @p name bla bla`. (So, not just simply prefixing the name with `@`)